src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntryAsync.File.Tests.Unix.cs

```diff
@@ -149,5 +149,40 @@ await using (TarReader reader = new TarReader(archive))
                 }
             }, format.ToString(), new RemoteInvokeOptions { RunAsSudo = true }).Dispose();
         }
+        
+        [ConditionalFact(nameof(IsRemoteExecutorSupportedAndPrivilegedProcess))]
+        public void CreateEntryFromFileOwnedByNonExistentGroup_Async()
+        {
+            RemoteExecutor.Invoke(async () =>
+            {
+                string groupName = Path.GetRandomFileName()[0..6];
+                int groupId = CreateGroup(groupName);
+
+                using TempDirectory root = new TempDirectory();
+
+                string fileName = "file.txt";
+                string filePath = Path.Join(root.Path, fileName);
+                File.Create(filePath).Dispose();
+
+                SetGroupAsOwnerOfFile(groupName, filePath);
+
+                DeleteGroup(groupName);
+
+                await using MemoryStream archive = new MemoryStream();
+                await using (TarWriter writer = new TarWriter(archive, TarEntryFormat.Ustar, leaveOpen: true))
+                {
+                    await writer.WriteEntryAsync(filePath, fileName); // Should not throw
+                }
+                archive.Seek(0, SeekOrigin.Begin);
+                
+                await using (TarReader reader = new TarReader(archive, leaveOpen: false))
+                {
+                    UstarTarEntry entry = await reader.GetNextEntryAsync() as UstarTarEntry;
+                    Assert.NotNull(entry);
+                    Assert.Equal(entry.GroupName, string.Empty);
```

Comment:
> Is this what other tar tools do?

Yes. Here's an example:

```
~/tartest $ touch file.txt

~/tartest $ ls -l file.txt
-rw-r--r-- 1 carlos carlos     0 Jan 24 10:50 file.txt

~/tartest $ sudo groupadd deleteme

~/tartest $ getent group deleteme
deleteme:x:3581:

~/tartest $ sudo chgrp deleteme file.txt

~/tartest $ ls -l file.txt
-rw-r--r-- 1 carlos deleteme     0 Jan 24 10:50 file.txt

~/tartest $ sudo groupdel deleteme

~/tartest $ ls -l file.txt
-rw-r--r-- 1 carlos   3581     0 Jan 24 10:50 file.txt

~/tartest $ tar cvf group.tar file.txt
file.txt
```

Then when inspecting the archive, you can see the gname field (yellow) is empty:

![image](https://user-images.githubusercontent.com/1175054/214384732-de24345c-8ed9-4e63-b3a0-8cafd38cdd66.png)

> I wonder if one of them chooses to write the groupName regardless of being non-existent.

But if the groupname is not known, we cannot write one, and we shouldn't be writing anything else, or we would be tampering with the original data we are archiving.

We only know the gid (it's in its own field in the header). We tried but failed to retrieve the groupname associated to that gid, because it does not exist in the current system, so we can just leave it empty.

Reply:
Thanks for checking. Does that apply for the other POSIX formats we support? 
`tar cvf --format=ustar group.tar file.txt`
`tar cvf --format=gnu group.tar file.txt`

For the case where you manually build a PosixTarEntry and set [GroupName](https://learn.microsoft.com/dotnet/api/system.formats.tar.posixtarentry.groupname), do we write the value even if it doesn't exist?

Reply:
> Does that apply for the other POSIX formats we support?

Yes. Confirmed, exact same behavior with gnu and ustar: The groupname field is left empty.



> For the case where you manually build a PosixTarEntry and set GroupName, do we write the value even if it doesn't exist?

That case is not affected by this bug because whatever value you set to that property is the value that ends up written in the archive. It does not get validated against the groups available in the current OS.